utils/getRushThreshold.ts

```diff
@@ -0,0 +1,31 @@
+import { AnalyticsEvent } from "@/types/admin/events";
+
+const mandatoryEvents = ["Info Session 1", "Info Session 2"]
+const remainingEvents = ["Professional Panel", "Resume Night", "Social Event"]
+
+export const getRushThreshold = (events: { [key: string]: boolean }) => {
+  // Check if at least one mandatory event was attended
+  const attendedMandatory = mandatoryEvents.some((event) => events[event]);
+
+  // Count how many remaining events were attended
+  const attendedRemaining = remainingEvents.filter((event) => events[event]).length;
+
+  // Check if the rush threshold is met: at least 1 mandatory event and 2 other events
+  return attendedMandatory && attendedRemaining >= 2;
```

Comment:
Can we make the count `2` is a constant? Additionally, you use that logic on both functions so maybe you can abstract this by creating a helper function like `isRushThresholdMet(...)` - verbosity in this case would help understand the functionality better.

Reply:
Completely agree on both. I'll update `2` to be a constant, but Imma come back to the abstraction on a separate PR bc I'm a bit swamped rn lol (and I wanna fix the analytics processing in the backend anyways)